src/Hooks/Hook.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace Carnival\Hooks;
+
+use Carnival\Hooks\Event;
+use Illuminate\Support\Collection;
+
+class Hook {
```

Comment:
I don't love the "hook system" class being called `Hook`. A hook, conceptually, is what a developer would attach to an "event" to be kept in the loop for when the event occurs.

Perhaps `HookSystem`, `HookManager`, `HookController`, etc. Thoughts?

Reply:
I kind of like HookController. But I like just "Hook" because it is short and explicit. Typing out HookSystem, HookController, etc. multiple times will get redundant. But if I had to choose one, HookController sounds good.

Reply:
`Hook` is short and explicit, but *this* isn't a hook so that'd be a bad name. It's an orchestrator of hooks
My only hesitation with HookController, even though I suggested it lol, is that it sort of implies this is a controller in the MVC sense. What do you think?

Reply:
Yeah, fair point. Lets go with HookSystem then.